lib/gollum/app.rb

```diff
@@ -348,7 +348,7 @@ def wiki_new
         wiki.write_page(name, format, params[:content], commit_message, path)
 
         page_dir = settings.wiki_options[:page_file_dir].to_s
-        redirect to("/#{clean_url(::File.join(page_dir, path, encodeURIComponent(name)))}")
+        redirect to("/#{clean_url(::File.join(encodeURIComponent(page_dir), encodeURIComponent(path), encodeURIComponent(name)))}")
```

Comment:
Untested, but wouldn't it be more efficient to call `encodeURIComponent` once with the joined URL string?